portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,52 @@
 // Copyright 2019 Google LLC
 //
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
 //
 //     https://www.apache.org/licenses/LICENSE-2.0
 //
 // Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// random image generator
+window.onload = SetImageOnMouseOver;
 
 /**
- * Adds a random greeting to the page.
+ * Change the given image to a random one whenever the user hovers over
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function SetImageOnMouseOver() {
+  document.getElementById('rand-front-tile')
+      .addEventListener('mouseover', function() {
+        getRandomFile(
+            /* element= */ document.querySelector('#rand-back-tile img'),
+            /* path= */ 'images/random/', /* fileName= */ 'test',
+            /* numElems= */ 4)
+      });
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/**
+ *
+ * @param {string} element - which DOM element to modify
+ * @param {string} path - the path starting from the webapp/ directory that the
+ *     files are located in
+ * @param {string} fileName - the startPrefix of the files in this directory
+ *     (assumed all files of the form fileNameXX.jpg)
+ * @param {int} numElems - the number of elements in the given directory
+ */
+function getRandomFile(element, path, fileName, numElems) {
+  // It would be ideal if files didn't have to follow a certain naming
+  // convention, but not sure this is possible with strictly server side
+  // JavaScript. After conferring with help line seems this is indeed
+  // impossible.
+  // TODO(morleyd): Implement on server side to clean up need for numElems, etc
+  // in the directory For now copy directly the example code with minor
+  // modifications.
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  var randomFile =
+      path + fileName + (Math.floor(Math.random() * numElems) + 1) + '.jpg';
```

Comment:
Consider pulling out the math into a variable, then concatenating in a separate statement. That way the reader can double-check the math without getting distracted by filename stuff, and easily scan the string construction without getting distracted by math.